### PR TITLE
[Projects] Unreachable code

### PR DESCRIPTION
**Pull request by george0st**
Add information about unreachable code

### CODE REVIEW

File: mlrun/projects/project.py [hunk 1]
```diff
@@ -1517,6 +1517,7 @@ def get_artifact(spec):
                     artifact, local_path=temp_dir, artifact_path=artifact_path
                 )
 
+            # TODO - unreachable (weak) code (see return)
```
Comment (liranbg) [inline mlrun/projects/project.py:1520]: looks like a mistake. @yaronha can confirm butregardless, can be removed (and not TODO'ed)

Comment (george0st) [reply mlrun/projects/project.py:1520]: @liranbg,  I hoped, that note will be remove this part of code or move it before return ... let me know.

Comment (yaronha) [reply mlrun/projects/project.py:1520]: look like line 1520 should move up, before the return and after 1503 (i.e. delete the local copy of the zip file when the data source is a remote object (not file))

**Comment by george0st** [comment]
@yaronha, I made small change (Only a '*.zip' file will be processed, not the directory with name '*.zip', it can generate exception during 'zipfile.extractall'). Let me know, if it is fine for you.

**Comment by liranbg** [comment]
Thanks @george0st  - I see that @alonmr  has found the same issue and covered it better - see https://github.com/mlrun/mlrun/pull/3323

**Pull request closed without merging**